### PR TITLE
docs: add demo overview to README

### PR DESCRIPTION
**Pull request by alexashley**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,1118 @@
 # gh-trusted-builds-app
+
+An example of using GitHub Actions [reusable workflows](https://docs.github.com/en/actions/using-workflows/reusing-workflows) to create and deploy a trusted software artifact.
+
+## Background
+
+These days, companies of all sizes and industries create software, from small scripts to complicated back-of-house inventory forecasting software.
```
Comment (rcoy-v) [inline README.md:7]:
```suggestion
These days, companies of all sizes and industries create software, from small scripts to complicated back-of-house inventory forecasting systems.
```

File: README.md [hunk 2]
```diff
@@ -1 +1,1118 @@
...
+- `ACTIONS_ID_TOKEN_REQUEST_URL`
+- `ACTIONS_RUNTIME_TOKEN`
+
+The job can then make an HTTP `POST` request to `$ACTIONS_ID_TOKEN_REQUEST_URL` with `$ACTIONS_RUNTIME_TOKEN` in the authorization header (using the [bearer token scheme](https://datatracker.ietf.org/doc/html/rfc6750#section-2.1)), and the job will receive an
+id token in the `value` field of the JSON response.
+
+For a more in-depth overview of using GitHub Actions' OIDC feature with reusable workflows, the SLSA GitHub Generator [specifications](https://github.com/slsa-framework/slsa-github-generator/blob/main/SPECIFICATIONS.md)
+go into additional detail.
+
+#### Sigstore: Cosign & Policy Controller
```
Comment (rcoy-v) [inline README.md:177]: I think these can be separate sections.

File: README.md [hunk 3]
```diff
@@ -1 +1,1118 @@
...
+1. Fulcio will validate the id token and check the provided proof
+1. If the request is valid, Fulcio will issue a code-signing certificate and populate the certificate fields with information from the id token
+   - When using a GitHub Actions OIDC token, the issuer will be `https://token.actions.githubusercontent.com` and the subject will be the value of the `job_workflow_ref` claim
+1. Depending on the deployment type, Fulcio will also submit a request to a certificate transparency log to include the certificate in the logs. The response from the certificate transparency log will include a signed timestamp that will be added to the certificate.
+
+After receiving the certificate, `cosign` or other tools can sign images or attestations with the private key generated earlier, and then upload the signature and certificate to Rekor.
+Once the signing is finished, the key pair can be discarded, as the certificate will be in Rekor for validation later.
+
+For a more concrete example, this is a sequence diagram that shows how a GitHub Actions workflow might build & sign a container image:
+```mermaid
```
Comment (rcoy-v) [inline README.md:213]: This diagram is really useful, but I think GitHub's rendering of larger mermaid documents is still too lacking. It's not easy to navigate and zoom on this when trying to understand the contents. Could we do something like generate a svg from the mermaid diagram, and include that asset in the readme?

Comment (alexashley) [reply README.md]: I converted to svg with the Mermaid CLI. The inline view is about the same, but it's easier to open in a new tab and see the full diagram

**Review by rcoy-v — APPROVED** [review]

**Pull request merged**